dev/core/src/com/google/gwt/dev/jjs/ast/JCaseStatement.java

```diff
@@ -17,28 +17,68 @@
 
 import com.google.gwt.dev.jjs.SourceInfo;
 
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Java case statement.
  */
 public class JCaseStatement extends JStatement {
 
-  private JExpression expr;
+  private List<JExpression> exprs;
 
   public JCaseStatement(SourceInfo info, JExpression expr) {
     super(info);
-    this.expr = expr;
+    this.exprs = Collections.singletonList(expr);
+  }
+
+  public JCaseStatement(SourceInfo info, Collection<JExpression> exprs) {
+    super(info);
+    this.exprs = Collections.unmodifiableList(new ArrayList<>(exprs));
   }
 
+  public boolean isDefault() {
+    return exprs.isEmpty();
+  }
+
+  @Deprecated
   public JExpression getExpr() {
-    return expr;
+    if (exprs.size() > 1) {
+      throw new IllegalStateException(
+              "JCaseStatement.getExpr() called on a node with multiple expressions " + exprs);
+    }
+    return exprs.isEmpty() ? null : exprs.get(0);
+  }
+
+  public List<JExpression> getExprs() {
+    return exprs;
+  }
```

Comment:
It does not seems like this is being used anywhere now in gwt itself,  but if this is used internally by the compiler then why not just remove it instead of deprecation? is there any external usage for this?